code/game/machinery/big_manipulator/big_manipulator.dm

```diff
@@ -0,0 +1,909 @@
+		diag_hud.add_atom_to_hud(src)
+
+	update_hud()
+	update_strategies()
+
+/// Checks the component tiers, adjusting the properties of the manipulator.
+/obj/machinery/big_manipulator/proc/process_upgrades()
+	var/datum/stock_part/servo/locate_servo = locate() in component_parts
+	if(!locate_servo)
+		return
+
+	var/manipulator_tier = locate_servo.tier
+	switch(manipulator_tier)
+		if(-INFINITY to 1)
+			min_speed_multiplier = MIN_SPEED_MULTIPLIER_TIER_1
+			max_speed_multiplier = MAX_SPEED_MULTIPLIER_TIER_1
+			interaction_point_limit = MAX_INTERACTION_POINTS_TIER_1
+			set_greyscale(COLOR_YELLOW)
+			manipulator_arm?.set_greyscale(COLOR_YELLOW)
+		if(2)
+			min_speed_multiplier = MIN_SPEED_MULTIPLIER_TIER_2
+			max_speed_multiplier = MAX_SPEED_MULTIPLIER_TIER_2
+			interaction_point_limit = MAX_INTERACTION_POINTS_TIER_2
+			set_greyscale(COLOR_ORANGE)
+			manipulator_arm?.set_greyscale(COLOR_ORANGE)
+		if(3)
+			min_speed_multiplier = MIN_SPEED_MULTIPLIER_TIER_3
+			max_speed_multiplier = MAX_SPEED_MULTIPLIER_TIER_3
+			interaction_point_limit = MAX_INTERACTION_POINTS_TIER_3
+			set_greyscale(COLOR_RED)
+			manipulator_arm?.set_greyscale(COLOR_RED)
+		if(4 to INFINITY)
+			min_speed_multiplier = MIN_SPEED_MULTIPLIER_TIER_4
+			max_speed_multiplier = MAX_SPEED_MULTIPLIER_TIER_4
+			interaction_point_limit = MAX_INTERACTION_POINTS_TIER_4
+			set_greyscale(COLOR_PURPLE)
+			manipulator_arm?.set_greyscale(COLOR_PURPLE)
+
+	active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION * BASE_POWER_USAGE * manipulator_tier
+
+/obj/machinery/big_manipulator/examine(mob/user)
+	. = ..()
+	var/mob/monkey_resolve = monkey_worker?.resolve()
+	if(!isnull(monkey_resolve))
+		. += "You can see a poor [monkey_resolve.name] buckled to [src]. You wonder if it's getting paid enough."
+
+/obj/machinery/big_manipulator/attack_hand_secondary(mob/living/user, list/modifiers)
+	try_press_on(user)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/machinery/big_manipulator/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+
+	context[SCREENTIP_CONTEXT_RMB] = "Toggle"
+
+	if(isnull(held_item))
+		context[SCREENTIP_CONTEXT_LMB] = panel_open ? "Interact with wires" : "Open UI"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	if(held_item.tool_behaviour == TOOL_WRENCH)
+		context[SCREENTIP_CONTEXT_LMB] = "[anchored ? "Una" : "A"]nchor"
+		context[SCREENTIP_CONTEXT_RMB] = "Rotate clockwise"
+		return CONTEXTUAL_SCREENTIP_SET
+	if(held_item.tool_behaviour == TOOL_SCREWDRIVER)
+		context[SCREENTIP_CONTEXT_LMB] = "[panel_open ? "Close" : "Open"] panel"
+		return CONTEXTUAL_SCREENTIP_SET
+	if(held_item.tool_behaviour == TOOL_CROWBAR && panel_open)
+		context[SCREENTIP_CONTEXT_LMB] = "Deconstruct"
+		return CONTEXTUAL_SCREENTIP_SET
+	if(is_wire_tool(held_item) && panel_open)
+		context[SCREENTIP_CONTEXT_LMB] = "Interact with wires"
+		return CONTEXTUAL_SCREENTIP_SET
+
+/obj/machinery/big_manipulator/Destroy(force)
```

Comment:
it's good practice to keep destroy right next to initialize, helps avoid missing cleanup in setup.